tutorials/groups.qmd

```diff
@@ -0,0 +1,242 @@
+xθi = π/6
+```
+
+### From Coordinates
+
+To get our first Lie algebra element we can use the textbook [`hat`](@ref), or equivaliently a more generalized [`get_vector`](@ref), function:
+
+```{julia}
+X_ = hat(G, IG, xθi)              # specific definition to Lie groups
+xXi = get_vector(G, p0, xθi, B)   # generalized definition beyond Lie groups
+println(xXi)
+@assert isapprox(X_, xXi)
+```
+
+Note that `hat` here assumes a default (orthogonal) basis for the more general `get_vector`.
+
+::: {.callout-note}
+In this case, the same would work given the base manifold [`Rotations(2)`](@ref):
+```{julia}
+_X_ = hat(M, p0, xθi)             # Lie groups definition
+_X = get_vector(M, p0, xθi, B)   # generalized definition
+@assert _X_ == xXi; @assert _X == xXi
+```
+One more caveat here is that for the Rotation matrices, the tangent vectors are always stored as elements from the Lie algebra.
+:::
+
+Now, let's place this algebra element on the manifold using the exponential map [`exp`](@ref):
+```{julia}
+xRi = exp(G, p0, xXi)
+# similarly for known underlying manifold
+xRi_ = exp(M, p0, xXi)
+
+@assert isapprox(xRi, xRi_)
+```
+
+### To Coordinates
+
+The logarithm transform from the group back to algebra (or coordinates) is:
+```{julia}
+xXi_ = log(G, p0, xRi)
+xXi__ = log(M, p0, xRi)
+@assert xXi ≈ xXi__
+```
+
+Similarly, the coordinate value can be extracted from the algebra using [`vee`](@ref), or directly from the group using the more generalized [`get_coordinates`](@ref):
+
+```{julia}
+# extracting coordinates using vee
+xθi__ = vee(G, p0, xXi_)[1]
+_xθi__ = vee(M, p0, xXi_)[1]
+
+# OR, the preferred generalized get_coordinate function
+xθi_ = get_coordinates(G, p0, xXi_, B)[1]
+_xθi_ = get_coordinates(M, p0, xXi_, B)[1]
+
+# confirm all versions are correct
+@assert isapprox(xθi, xθi_); @assert isapprox(xθi, _xθi_)
+@assert isapprox(xθi, xθi__); @assert isapprox(xθi, _xθi__)
+```
+
+::: {.callout-note}
+The disadvantage might be that the representation of `X` is not nice, i.e. it uses too much space or doing vector-calculations is not so easy.  E.g. fixed rank matrices are overloaded for all vector operations, but maybe that is “not enough” for a general user application that really wants vectors. But: Given a basis `B` one can look at the coefficients of the tangent vector `X` with respect to basis `B`.  From the Sphere example note above the basis vectors would be `Y1=[0.0, 1.0, 0.0]` and `Y2=[0.0, 0.0, 1.0]`, so to get the coordinates would be `c = get_coordinates(Sphere(2), p, X, B)`.  Vice versa, if you have a coordinate vector with respect to a basis `B` of the tangent space at `p` and want the vector back, then you do `X2 = get_vector(M, p, c, B)` (and you should have `X2==X`).  The coordinate vector `c` might also have the advantage of saving memory. For example SPD matrix tangent vectors take $n^2$ entries to save, i.e. storing the full matrix, but the coordinate vectors only take $\frac{n(n+1)}{2}$.
```

Comment:
This is a bit long and I do no understand what this is for. Sphere example above? also a short reference to fixed rank is unclear to me why this is here.

Reply:
I wasn't sure about this note either so maybe let's remove it?

Reply:
Yes, I think until we understand more closely what disadvantage is meant, we should remove this.